modules/index-rebuild/src/main/java/org/opencastproject/index/rebuild/IndexRebuildService.java

```diff
@@ -0,0 +1,228 @@
+/**
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.index.rebuild;
+
+import static java.lang.String.format;
+
+import org.opencastproject.search.index.AbstractSearchIndex;
+
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceListener;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class IndexRebuildService implements BundleActivator {
+
+  /**
+   * The services whose data is indexed by ElasticSearch.
+   * Attention: The order is relevant for the index rebuild and should not be changed!
+   */
+  public enum Service {
+    Groups, Acl, Themes, Series, Scheduler, Workflow, AssetManager, Comments
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(IndexRebuildService.class);
+  private final Map<IndexRebuildService.Service, IndexProducer> indexProducers = new HashMap<>();
```

Comment:
Not thread-safe. Use a ConcurrentHashMap

Reply:
Good point, will do.